app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -40,6 +40,14 @@ def build_enumerator(_run, cursor:)
         raise ArgumentError, "#{@task.class.name}#collection must be either "\
           "an Active Record Relation, Array, or CSV."
       end
+
+      @task.throttle_specs.reduce(collection_enumerator) do |enumerator, spec|
```

Comment:
Maybe we could extract that logic to a method?

Reply:
I feel like it's pretty readable as is - if I extracted it to a method I'd probably name if `reduced_throttle_conditions`, which seems to be equally legible with what's there. Feel free to push back if you feel strongly 😄